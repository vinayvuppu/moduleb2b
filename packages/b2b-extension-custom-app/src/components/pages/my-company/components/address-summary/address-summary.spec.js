import React from 'react';
import { shallow } from 'enzyme';
import AddressSummary from './address-summary';

const createTestProps = props => ({
  address: {
    firstName: 'John',
    lastName: 'Doe',
    phone: '1234568',
    email: 'foo@bar.com',
    company: 'commercetools',
    city: 'Munich',
    region: 'Bavaria',
    country: 'DE',
  },
  ...props,
});

const addressSimpleFields = [
  {
    field: 'firstName',
    value: 'John',
    index: 12,
  },
  {
    field: 'lastName',
    value: 'Doe',
    index: 13,
  },
  {
    field: 'company',
    value: 'commercetools',
    index: 16,
  },
  {
    field: 'city',
    value: 'Munich',
    index: 18,
  },
  {
    field: 'postalCode',
    value: '12345',
    index: 19,
  },
  {
    field: 'additionalAddressInfo',
    value: 'Some additional address info',
    index: 23,
  },
];

describe('render', () => {
  let props;
  let wrapper;
  beforeEach(() => {
    props = createTestProps();
    wrapper = shallow(<AddressSummary {...props} />);
  });
  it('should output snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  addressSimpleFields.forEach(({ field, value, index }) => {
    describe(`${field}`, () => {
      describe('when a value is set', () => {
        beforeEach(() => {
          props = createTestProps({
            address: { [field]: value, country: 'DE' },
          });
          wrapper = shallow(<AddressSummary {...props} />);
        });
        it(`should render the ${field} value`, () => {
          expect(wrapper.find('TextDetail').at(index)).toHaveProp(
            'children',
            value
          );
        });
      });
      describe('when no value is set', () => {
        beforeEach(() => {
          props = createTestProps({
            address: { [field]: undefined, country: 'DE' },
          });
          wrapper = shallow(<AddressSummary {...props} />);
        });
        it('should render the hyphen (`-`) as a fallback value', () => {
          expect(wrapper.find('TextDetail').at(index)).toHaveProp(
            'children',
            '-'
          );
        });
      });
    });
  });
  describe('address', () => {
    describe('when no street and no street number available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: {
            streetName: undefined,
            streetNumber: undefined,
            country: 'DE',
          },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should render the hyphen (`-`) value', () => {
        expect(wrapper.find('TextDetail').at(17)).toHaveProp('children', '-');
      });
    });

    describe('when only street name available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: {
            streetName: 'street road',
            streetNumber: undefined,
            country: 'DE',
          },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should return the street name as address value', () => {
        expect(wrapper.find('TextDetail').at(17)).toHaveProp(
          'children',
          'street road'
        );
      });
    });
    describe('when street and street number available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: {
            streetName: 'street road',
            streetNumber: '155',
            country: 'DE',
          },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should return the street name and number as address value', () => {
        expect(wrapper.find('TextDetail').at(17)).toHaveProp(
          'children',
          'street road 155'
        );
      });
    });
  });

  describe('region/state', () => {
    describe('when no region and no state available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: { region: undefined, state: undefined, country: 'DE' },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should render the hyphen (`-`) value', () => {
        expect(wrapper.find('TextDetail').at(20)).toHaveProp('children', '-');
      });
    });

    describe('when region available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: {
            region: 'Comunitat Valenciana',
            state: undefined,
            country: 'DE',
          },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should return the region value', () => {
        expect(wrapper.find('TextDetail').at(20)).toHaveProp(
          'children',
          'Comunitat Valenciana'
        );
      });
    });

    describe('when state available', () => {
      beforeEach(() => {
        props = createTestProps({
          address: { region: undefined, state: 'Texas', country: 'DE' },
        });
        wrapper = shallow(<AddressSummary {...props} />);
      });
      it('should return the state value', () => {
        expect(wrapper.find('TextDetail').at(20)).toHaveProp(
          'children',
          'Texas'
        );
      });
    });
  });
});
