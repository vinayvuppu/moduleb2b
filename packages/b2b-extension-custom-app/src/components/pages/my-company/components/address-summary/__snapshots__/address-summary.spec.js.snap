// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should output snapshot 1`] = `
<Inline
  alignItems="flex-start"
  justifyContent="flex-start"
  scale="m"
>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Name"
        description="The label for the first name"
        id="Orders.Create.Step.ShippingMethod.addressFirstNameLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Last name"
        description="The label for the last name"
        id="Orders.Create.Step.ShippingMethod.addressLastNameLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Phone"
        description="The label for the phone"
        id="Orders.Create.Step.ShippingMethod.addressPhoneLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Email"
        description="The label for the email"
        id="Orders.Create.Step.ShippingMethod.addressEmailLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Company name"
        description="The label for the company name"
        id="Orders.Create.Step.ShippingMethod.addressCompanyNameLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Address"
        description="The label for the street"
        id="Orders.Create.Step.ShippingMethod.addressStreetLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="City"
        description="The label for the city"
        id="Orders.Create.Step.ShippingMethod.addressCityLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Postal Code"
        description="The label for the postal code"
        id="Orders.Create.Step.ShippingMethod.addressPostalCodeLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Region"
        description="The label for the region"
        id="Orders.Create.Step.ShippingMethod.addressRegionLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Country"
        description="The label for the country"
        id="Orders.Create.Step.ShippingMethod.addressCountryLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Additional street information"
        description="The label for the additional street information"
        id="Orders.Create.Step.ShippingMethod.addressAdditionalStreetInformationLabel"
        values={Object {}}
      />
    </TextDetail>
    <TextDetail
      isBold={true}
    >
      <FormattedMessage
        defaultMessage="Additional information"
        description="The label for the additional information"
        id="Orders.Create.Step.ShippingMethod.addressAdditionalInformationLabel"
        values={Object {}}
      />
    </TextDetail>
  </Stack>
  <Stack
    alignItems="stretch"
    scale="s"
  >
    <TextDetail>
      John
    </TextDetail>
    <TextDetail>
      Doe
    </TextDetail>
    <TextDetail>
      1234568
    </TextDetail>
    <TextDetail>
      foo@bar.com
    </TextDetail>
    <TextDetail>
      commercetools
    </TextDetail>
    <TextDetail>
      -
    </TextDetail>
    <TextDetail>
      Munich
    </TextDetail>
    <TextDetail>
      -
    </TextDetail>
    <TextDetail>
      Bavaria
    </TextDetail>
    <TextDetail>
      DE
    </TextDetail>
    <TextDetail>
      -
    </TextDetail>
    <TextDetail>
      -
    </TextDetail>
  </Stack>
</Inline>
`;
